File: python/functions/function_calling_function.py
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
# SPDX-License-Identifier: GPL-3.0-or-later

import typing

def inside() -> str:
    print("* Inside.")
    return "IN"

def outside() -> str:
    print("# Outside:\n"
        + "# Calling function from function...")
    result = inside()
    print(f"# result: {result}")
    return "OUT"

print("Calling function...")
result = outside()
print(f"result: {result}")

File: javascript/functions/function_calling_function.js
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

function inside() {
  console.log("* Inside.");
  return "IN";
}

function outside() {
  console.log("# Outside:\n"
    + "# Calling function from function...");
  let result = inside();
  console.log("# result: " + result);
  return "OUT";
}

console.log("Calling function...");
let result = outside();
console.log("result: " + result);
